test/app-tests/loading.js

```diff
@@ -493,6 +494,76 @@ describe('loading:', function () {
                 expect(windowLocation.hash).toEqual("#/room/!room:id");
             }).done(done, done);
         });
+
+        describe('Login as user', function() {
+            beforeEach(function() {
+                // first we have to load the homepage
+                loadApp();
+
+                httpBackend.when('POST', '/register').check(function(req) {
+                    expect(req.queryParams.kind).toEqual('guest');
+                }).respond(200, {
+                    user_id: "@guest:localhost",
+                    access_token: "secret_token",
+                });
+
+                return httpBackend.flush().then(() => {
+                    return awaitSyncingSpinner(matrixChat);
+                }).then(() => {
+                    // we got a sync spinner - let the sync complete
+                    return expectAndAwaitSync();
+                }).then(() => {
+                    // once the sync completes, we should have a home page
+                    ReactTestUtils.findRenderedComponentWithType(
+                        matrixChat, sdk.getComponent('structures.HomePage'));
+
+                    // we simulate a click on the 'login' button by firing off
+                    // the relevant dispatch.
+                    //
+                    // XXX: is it an anti-pattern to access the react-sdk's
```

Comment:
We don't _really_ need to test the login button before every test, I think it'd be best to use `start_login` given that this is the `beforeEach`. If we had a dedicated integration test for the flow where the user clicks on "Login", logs in etc., then it'd be better to find the button and simulate a click.